### PR TITLE
Fix labels and unit selection in widgets

### PR DESCRIPTION
**Pull request by alejoe91**
Previously, the `plot_unit_locations` and `plot_spike_locations` were plotting a legend with ALL units (in grey) and only selected units in color. This clealy doesn't scale up with number of units.

Now only selected units are displayed in the legend. In addition, for the `ipywidgets` backend, the first available unit is selected at initiation.

Finally, added `max_spikes_per_unit` argument in `plot_spike_locations()` and `plot_amplitudes` to keep the plots manageable with long recordings and many units

### CODE REVIEW

File: spikeinterface/widgets/ipywidgets/amplitudes.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def do_plot(self, data_plot, **backend_kwargs):
                 fig = plt.figure(figsize=((ratios[1] * width_cm) * cm, height_cm * cm))
                 plt.show()
 
+        data_plot['unit_ids'] = [data_plot['unit_ids'][0]]
```
Comment (samuelgarcia) [inline spikeinterface/widgets/ipywidgets/amplitudes.py:37]: This is very very weird weird and hard to understand.
This sounds like a design problem. No ?

File: spikeinterface/widgets/template_similarity.py [hunk 2]
```diff
@@ -41,6 +44,9 @@ def __init__(self, waveform_extractor: WaveformExtractor,
             unit_indices = sorting.ids_to_indices(unit_ids)
             similarity = similarity[unit_indices][:, unit_indices]
 
+        if not display_diagonal_values:
+            np.fill_diagonal(similarity, 0)
```
Comment (samuelgarcia) [inline spikeinterface/widgets/template_similarity.py:48]: I think I really prefer this solution rather than hacking the computation function itself .

**Comment by samuelgarcia** [comment]
This OK for me except the ` data_plot['unit_ids'] = [data_plot['unit_ids'][0]]` which is weird.

File: spikeinterface/widgets/ipywidgets/amplitudes.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/widgets/ipywidgets/amplitudes.py:37]: No, the problem is that by default when `unit_ids` is None, all of them are selected. With the `ipywidgets` backend is better to initialize with one unit, and the user can select anything later. 

If you have hundreds of units it doesn't make sense to plot everything at once (but that's default)

File: spikeinterface/widgets/amplitudes.py [hunk 3]
```diff
@@ -17,15 +19,25 @@ class AmplitudesWidget(BaseWidget):
...
+        If None (default), this is automatically adjusted.
     """
     possible_backends = {}
 
     
     def __init__(self, waveform_extractor: WaveformExtractor, unit_ids=None, unit_colors=None,
-                 segment_index=None, hide_unit_selector=False, plot_histograms=False,
-                 bins=None, backend=None, **backend_kwargs):
+                 segment_index=None, max_spikes_per_unit=500, hide_unit_selector=False, 
+                 plot_histograms=False, bins=None, backend=None, **backend_kwargs):
```
Comment (samuelgarcia) [inline spikeinterface/widgets/amplitudes.py:40]: I think for amplitudes drift checking having all is still good.
Maybe we could have num_spike_per_seconds.

File: spikeinterface/widgets/spike_locations.py [hunk 4]
```diff
@@ -64,10 +68,21 @@ def __init__(
...
             unit_ids = sorting.unit_ids
 
+        all_spike_locs = spike_locations[segment_index]
+        if max_spikes_per_unit is None:
+            spike_locs = all_spike_locs
+        else:
+            spike_locs = dict()
+            for unit, locs_unit in all_spike_locs.items():
+                if len(locs_unit) > max_spikes_per_unit:
+                    spike_locs[unit] = locs_unit[np.random.permutation(len(locs_unit))[:max_spikes_per_unit]]
```
Comment (samuelgarcia) [inline spikeinterface/widgets/spike_locations.py:78]: I think I prefer the 
```inds = np.random.choice(total, size=max_spikes_per_unit, replace=False)``` writting for this.
And I think it is more efficient.

File: spikeinterface/widgets/matplotlib/unit_locations.py [hunk 5]
```diff
@@ -56,20 +56,20 @@ def do_plot(self, data_plot, **backend_kwargs):
...
         for p in patches:
             self.ax.add_patch(p)
-            
         handles = [Line2D([0], [0], ls="", marker='o', markersize=5, markeredgewidth=2, 
-                          color=unit_colors[unit]) for unit in unit_ids]
+                          color=unit_colors[unit]) for unit in dp.unit_ids]
             
         self.figure.legend(handles, labels, loc='upper center', bbox_to_anchor=(0.5, 1.),
                            ncol=5, fancybox=True, shadow=True)
+        self.ax.axis("off")
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply spikeinterface/widgets/matplotlib/unit_locations.py]: added `hide_axis` option

**Comment by alejoe91** [comment]
@samuelgarcia ready to go on my side

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**